src/components/organisms/dialog-preview-image/index.js

```diff
@@ -1,25 +1,49 @@
-import React from 'react'
+import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 
 import 'react-image-crop/dist/ReactCrop.css'
 
 import ImageCrop from '_molecules/image-crop'
 import Dialog from '_molecules/dialog'
 import Image from '_atoms/image'
+import Button from '_atoms/button'
 
-const DialogPreviewImage = ({ invalidProperties, selectedFile, crop, withCrop, ...props }) => (
-  <Dialog {...props}>
-    {withCrop ? (
-      <ImageCrop selectedFile={selectedFile} crop={crop} />
-    ) : (
-      <Image image={selectedFile} />
-    )}
-  </Dialog>
-)
+class DialogPreviewImage extends Component {
+  state = {
+    newImage: this.props.selectedFile,
+  }
+
+  setNewImage = image => {
+    this.setState({ newImage: image })
+  }
+
+  render() {
+    const { newImage } = this.state
+    const {
+      invalidProperties,
+      selectedFile,
+      crop,
+      withCrop,
+      onContinueClick,
+      ...props
+    } = this.props
+    return (
+      <Dialog {...props}>
+        {withCrop ? (
+          <ImageCrop selectedFile={selectedFile} crop={crop} onImageCropped={this.setNewImage} />
+        ) : (
+          <Image image={selectedFile} />
+        )}
+        <Button text="Save" onClick={() => onContinueClick(newImage)} />
```

Comment:
avoid anonymous function.

Reply:
Yeap, but in this case, the function receives a parameter. I don't know a better way if you know let me know :)

Reply:
```jsx
const onSave =() => {
  onContinueClick(newImage)
}
<Button text="Save" onClick={onSave} />
```